geopandas/geodataframe.py

```diff
@@ -252,9 +252,17 @@ def set_geometry(self, col, drop=False, inplace=False, crs=None):
 
         Parameters
         ----------
-        col : column label or array
+        col : column label or array-like
+            An existing column name or values to set as the new geometry column.
+            If values (array-like, Series) are passed, then if they are named (Series)
+              the new geometry column will have the corresponding name, otherwise
```

Comment:
Maybe something like "a new column will be added with that name and set as the new geometry column"? (to explicitly say that the resulting GeoDataFrame has an additional column, compared to the case of the values of the existing column being replaced)

Reply:
It isn't necesarily always an additional column though, if a column with that name already exists. Wordy, but perhaps saying "...the new geometry column will have the same name (newly added to the GeoDataFrame if it was not already a column)..."